app/code/Magento/Quote/Plugin/UpdateCartId.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\Quote\Plugin;
+
+use Magento\Framework\Webapi\Rest\Request as RestRequest;
+use Magento\Quote\Api\Data\CartItemInterface;
+use Magento\Quote\Api\GuestCartItemRepositoryInterface;
+
+/**
+ * Update cart id from request param
+ */
+class UpdateCartId
+{
+    /**
+     * @var RestRequest $request
+     */
+    private $request;
+
+    /**
+     * @param RestRequest $request
+     */
+    public function __construct(RestRequest $request)
+    {
+        $this->request = $request;
+    }
+
+    /**
+     * Update id from request if param cartId exist
+     *
+     * @param GuestCartItemRepositoryInterface $guestCartItemRepository
+     * @param CartItemInterface $cartItem
+     * @return void
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function beforeSave(
+        GuestCartItemRepositoryInterface $guestCartItemRepository,
+        CartItemInterface $cartItem
+    ): void {
+        if ($cartId = $this->request->getParam('cartId')) {
```

Comment:
Hello @engcom-Charlie, I also used to implement assignments within comparisons but one day someone pointed me out that it can be a bit misleading: is the `=` a real assignment or a bug, that is, a missing `==`? 

Reading the code without too much attention (it happens) it may seem a bug, someone could be tempted to correct it causing indeed a bug :)

At that point, I stopped using it and switched back to assigning the variable and doing the comparison in two different statements.

Would you consider doing that change, thus joining my group of old, wise, and defensive programmers like me? :-)